### DISCUSSION TITLE
Tasmota Thermostat - Control Response

### DISCUSSION THREAD

**Discussion by erniberni** [arendst/Tasmota#17538] [Show and tell]
Hi,
I am working on a project with Tasmota Thermostat. 
A small container should be kept on constant temperature which is working up to now quite well.
I am only struggling with the behavior when initial switched on an a big temperature difference must be controlled.
The thermostat swings far above the preset temperature and after that the temperature is controlled very well.
How can I prevent this behavior?
Settings different from default are:
TIMEPICYCLESET 10
PROPBANDSET 4
TIMEMINACTIONSET 1
Greetings from Cologne
Reinhard
![Screenshot_20221229_151600](https://user-images.githubusercontent.com/1084946/209965916-047dfc1e-0e33-4712-9eaa-ecd978364589.png)

**Comment by barbudor** [comment]
I'm not using thermostat myself but there are some elements there https://tasmota.github.io/docs/Thermostat/#pi-controller-main-parameters

I feel that the cycle should be reduced, at least for some time after a new target has been given

**Comment by Wintespe** [comment]
It looks as if the parameters (I component) have to be adjusted
https://blog.clanlaw.org.uk/pid-loop-tuning.html
